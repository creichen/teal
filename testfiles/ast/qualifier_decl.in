qualifier Qual1 [t1, t2] (a1, a2 : t2) (x : t) = {}

qualifier Qual2 (a1, a2 : t2) (x : t) = {}

qualifier Qual3 [t1, t2] (x, y) (x) <: Qual4, Qual5 = {}

qualifier Qual4 [t1, t2] () (x) <: Qual4, Qual5 = {}

// this won't work
// qualifier Qual4 [t1, t2] (x) <: Qual4, Qual5 = {}
