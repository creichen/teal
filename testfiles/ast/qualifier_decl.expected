Program
  List
  List
    QualifierDecl
      IdDecl ID="Qual1"
      List
        TypeVarDecl
          IdDecl ID="t1"
        TypeVarDecl
          IdDecl ID="t2"
      List
        VarDecl
          IdDecl ID="a1"
          Opt
          Opt
        VarDecl
          IdDecl ID="a2"
          Opt
            SimpleType
              IdUse ID="t2"
              List
          Opt
      VarDecl
        IdDecl ID="x"
        Opt
          SimpleType
            IdUse ID="t"
            List
        Opt
      List
      List
    QualifierDecl
      IdDecl ID="Qual2"
      List
      List
        VarDecl
          IdDecl ID="a1"
          Opt
          Opt
        VarDecl
          IdDecl ID="a2"
          Opt
            SimpleType
              IdUse ID="t2"
              List
          Opt
      VarDecl
        IdDecl ID="x"
        Opt
          SimpleType
            IdUse ID="t"
            List
        Opt
      List
      List
    QualifierDecl
      IdDecl ID="Qual3"
      List
        TypeVarDecl
          IdDecl ID="t1"
        TypeVarDecl
          IdDecl ID="t2"
      List
        VarDecl
          IdDecl ID="x"
          Opt
          Opt
        VarDecl
          IdDecl ID="y"
          Opt
          Opt
      VarDecl
        IdDecl ID="x"
        Opt
        Opt
      List
        SimpleType
          IdUse ID="Qual4"
          List
        SimpleType
          IdUse ID="Qual5"
          List
      List
    QualifierDecl
      IdDecl ID="Qual4"
      List
        TypeVarDecl
          IdDecl ID="t1"
        TypeVarDecl
          IdDecl ID="t2"
      List
      VarDecl
        IdDecl ID="x"
        Opt
        Opt
      List
        SimpleType
          IdUse ID="Qual4"
          List
        SimpleType
          IdUse ID="Qual5"
          List
      List
